modules/components/pages/inputs/pg_stream.adoc

```diff
@@ -0,0 +1,318 @@
+= pg_stream
+// tag::single-source[]
+:type: input
+:page-beta: true
+:categories: ["Services"]
+
+component_type_dropdown::[]
+
+Streams data changes from a PostgreSQL database using logical replication. There is also a configuration option to <<stream_snapshot,stream all existing data>> from the database.
+
+ifndef::env-cloud[]
+Introduced in version 4.40.0.
+endif::[]
+
+```yml
+# Configuration fields, showing default values
+input:
+  label: ""
+  pg_stream:
+    dsn: postgres://foouser:foopass@localhost:5432/foodb?sslmode=disable # No default (required)
+    batch_transactions: true
+    stream_snapshot: false
+    snapshot_memory_safety_factor: 1
+    snapshot_batch_size: 0
+    schema: public # No default (required)
+    tables: [] # No default (required)
+    checkpoint_limit: 1024
+    temporary_slot: false
+    slot_name: "" # No default (optional)
+    pg_standby_timeout: 10s
+    pg_wal_monitor_interval: 3s
+    max_parallel_snapshot_tables: 1
+    auto_replay_nacks: true
+    batching:
+      count: 0
+      byte_size: 0
+      period: "" # No default (optional)
+      check: "" # No default (optional)
+      processors: [] # No default (optional)
+```
+
+== Metadata
+
+This input adds the following metadata fields to each message:
+
+- `mode`: Whether a message is part of the snapshot processing of existing data in a database (`snapshot`) or a data update streamed to Redpanda Connect (`streaming`).
+- `table`: The name of the database table from which the message originated.
+- `operation`: The type of database operation that generated the message, such as `INSERT`, `UPDATE` or `DELETE`.
+
+== Fields
+
+=== `dsn`
+
+The data source name (DSN) of the PostgreSQL database from which you want to stream updates. Use the format `postgres://[user[:password]@][netloc][:port][/dbname][?param1=value1&...]`.
+
+By default, PostgreSQL enforces SSL. In secure environments, you can override this mode by appending the parameter `sslmode=disable` to the connection string.
+
+*Type*: `string`
+
+```yml
+# Examples
+
+dsn: postgres://foouser:foopass@localhost:5432/foodb
+
+dsn: postgres://foouser:foopass@localhost:5432/foodb?sslmode=disable
+```
+
+=== `batch_transactions`
+
+When set to `true`, transactions are batched into a single message.
```

Comment:
Why would users want to enable this option? What are they upsides and downsides?

Reply:
Reworded slightly.